src/resolvers-cassandra/Edges/queries.js

```diff
@@ -1,39 +1,43 @@
 'use strict';
 
 const Promise = require('promise');
+const flatten = require('lodash/flatten');
 const cassandraConnector = require('../../clients/cassandra/CassandraConnector');
 const featureServiceClient = require('../../clients/locations/FeatureServiceClient');
 const withRunTime = require('../shared').withRunTime;
 const trackEvent = require('../../clients/appinsights/AppInsightsClient').trackEvent;
 
 function makeSiteBboxQuery(args) {
   return {
-    query: 'SELECT geofence FROM sitesettings WHERE sitename = ?',
+    query: 'SELECT geofence FROM sitesettings WHERE sitename = ? ALLOW FILTERING',
     params: [args.site]
   };
 }
 
-function makeTermsQuery(args) {
-  let clauses = [];
-  let params = [];
-
-  if (args.fromDate) {
-    clauses.push('(event_time >= ?)');
-    params.push(args.fromDate);
-  }
-
-  if (args.toDate) {
-    clauses.push('(event_time <= ?)');
-    params.push(args.toDate);
-  }
-
-  if (args.sourceFilter && args.sourceFilter.length) {
-    clauses.push(`(pipeline IN (${args.sourceFilter.map(_ => '?').join(', ')}))`); // eslint-disable-line no-unused-vars
-    params = params.concat(args.sourceFilter);
-  }
-
-  let query = `SELECT detectedkeywords FROM fortis.events WHERE ${clauses.join(' AND ')}`;
-  return {query: query, params: params};
+function makeTermsQueries(args) {
+  const supportedPipelines = args.sourceFilter && args.sourceFilter.length ? args.sourceFilter : [null];
+  return supportedPipelines.map(pipeline => {
+    const clauses = [];
+    const params = [];
+
+    if (args.fromDate) {
+      clauses.push('(event_time >= ?)');
+      params.push(args.fromDate);
+    }
+
+    if (args.toDate) {
+      clauses.push('(event_time <= ?)');
+      params.push(args.toDate);
+    }
+
+    if (pipeline) {
+      clauses.push('(pipeline = ?)');
+      params.push(pipeline);
+    }
+
+    let query = `SELECT detectedkeywords FROM fortis.events WHERE ${clauses.join(' AND ')} ALLOW FILTERING`;
```

Comment:
Yes, the schema right now requires us to do this :( We really have to formulate the v2 queries that we want and make sure that the DB schema supports them. Adding @erikschlegel for visibility.